### PR TITLE
Add commands to rekey device and create packages

### PR DESCRIPTION
**Pull request by fumer-fubotv**
Added support for the following commands:

1. Rekey Device: Rekey specified device based on the signing password ("brightscript.remoteControl.signingPassword") and package file ("brightscript.remoteControl.signedPackagePath") provided.

2. Create Package: This will present user a list of available launch configs to choose from. Once selected it will create a .zip file and .pkg file based on the config and will save them  to the ${workspacefolder}/out

3. Rekey Device and Create Package: This will first rekey device and then create package for the selected launch config

### CODE REVIEW

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Hey, after I started looking over this, I realized we need to tweak the design a bit. We should sync up about this so I can explain in more detail. But we shouldn't use the "last used device" stuff like we did with the screenshot logic. Here's a rough overview of how the flow should work:

```
rekey: 
a) "How do you want to enter your stuff?"
 - "Pick from json file"
 - "Enter manually"
b) prompt for host (prepopulated from json above if possible)
c) Prompt for password (prepopulated from json above if possible)
d) Prompt for signed package path (prepopulated from json above if available)

package: 
a) "What should we package?"
 - "Pick a folder"
 - "Pick a zip"
 - "Pick from a launch.json"
    - show a launch.json option picker
 - "Pick a rokudeploy.json"
	- 

b) prompt for host (prepopulated from json above if possible)
c) Prompt for password (prepopulated from json above if possible)
d) Prompt for signed package path (prepopulated from json above if available)
e) Show a summary of stuff, ask to click "Yes, do it" or "No, something's wrong, cancel".

rekeyAndPackage: same as "package" but do the rekey first, no new input necessary

```

File: src/BrightScriptCommands.ts [hunk 1]
```diff
@@ -26,6 +29,8 @@ export class BrightScriptCommands {
     public host: string;
     public password: string;
     public workspacePath: string;
+    private signingPassword: string;
+    private signedPackagePath: string;
```
[the remark this answers is not in the document]
Comment (fumer-fubotv) [reply src/BrightScriptCommands.ts]: Updated. Please review

**Comment by fumer-fubotv** [comment]
> I have a few suggestions:
> 
> 1. This stuff in this file getting a bit large, so let's move all this new logic out of BrightScriptCommands.ts into a separate src/commands/RekeyAndPackageCommands.ts file.
> 2. There seem to be a few bugs around picking a json file for the "Rekey device" flow. When I pick a rokudeploy.json with only `rekeySignedPackage`, `devId`, `signingPassword`, `convertToSquashFS`, it fails with an error "A system error occurred (getaddrinfo ENOTFOUND :80). I'm assuming it's because the logic currently assumes that _everything_ was supposed to be included in the json. That's not how it should work. We should load any values we _find_ in the json, but then still walk through the manual prompts to allow the user to confirm those are the values they want, and fill in any missing values.
> 3. When running "Rekey Device", and choosing "enter manually". after I enter my signing password, it just pops up a new file picker dialog, but there was no messaging around WHAT I'm supposed to pick. I think we need another intermediary dropdown with instructions saying "pick your pkg" so they know _why_ the file picker appears.
> 4. After running the manual rekey for our internal app, I get this error at the end.
>    image

Most of the suggestions are addressed in the update. However i was not able to reproduce the error in #4 you mentioned

**Comment by TwitchBronBron** [comment]
@fumer-fubotv I finally got a chance to look over this. I've pushed some changes to optimize the flow a bit. 
The biggest change here is that we really need to prompt the user for all the details every time (either from json or from input boxes). We can't use values like the last password or last host used, because that might not be the device they want to rekey. 

One thing we need to tweak, is for the file picker modal, we should add information showing the currently selected zip (if there is one from rokudeploy.json), and add another button in that case saying "I want to change it". I see it having 2 flows:
1. there is no package picked yet, so they have "Pick a file" and "cancel". 
2. There is already a default package set, so they have "Use the current file", "Pick a different file", and "cancel"
image

It would be great if you could run through this workflow a few times to test it out, and iron out any rough edges. Then It should probably be good to go.

File: src/commands/RekeyAndPackageCommand.ts [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+            }
+
+            if (content.password) {
+                rekeyConfig.password = content.password;
+            }
+        }
+        return this.getRekeyManualEntries(rekeyConfig, rekeyConfig);
+    }
+
+    private async getRekeyManualEntries(rekeyConfig: RekeyConfig, defaultValues) {
```
Comment (TwitchBronBron) [inline src/commands/RekeyAndPackageCommand.ts:98]: `defaultValues` should have be of type `RekeyConfig` here

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Here are a few things to change:
 - when choosing "I want to change something", it's not showing the previously-selected host in the host picker. That `userInputManager.promptForHost()` function accepts a default value. We should pass in `rekeyConfig?host ?? defaultValues?.host`
image

- the "create package" flow doesn't remember the previously selected "what do you want to package" option when I choose "I want to change something". 
image

- the "rekey device and create package" command feels quirky. It seems like it just runs the rekey command and then the create package command next. It should collect all user input first (for rekey AND packaging inputs), and THEN show a summary at the end including all rekey inputs AND package/sign inputs. At that point, run both actions.

**Review by TwitchBronBron — COMMENTED** [review]
Found a few more issues. 
1. For "Create package" and choosing "use folder", it expects there to be a zip. But my folder has the source files, not a zip. We need to zip that folder as part of this flow. 
image

2. For "Create Package", when I choose "Use a rokudeploy.json" file, and if that file doesn't have `rootDir` specified, the packaging fails. 
image
image

3. For "Create package", when choosing "use launch.json", it appears to grab all the right values, but then the flow fails
image
image

**Comment by fumer-fubotv** [comment]
I tried to reproduce the scenarios mentioned above and below are the findings:
1. Create package using "use folder" works successfully if the specified folder has a manifest in it. Otherwise it will throw an error
2. Create package using "use a rokudeploy.json" expects source to be specified. 
3. Create Package using launch.json has been working fine for me. Lets connect to reproduce/resolve this issue

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**